Client/Frontend/Browser/Search/SearchEngines.swift

```diff
@@ -335,4 +338,91 @@ class SearchEngines {
             return index1! < index2!
         }
     }
+    
+    /// Function for Yahoo Search Engine Migration
+    /// If Default Search Engine is Yahoo or Yahoo! JAPAN,
+    /// the engine will be migrated as Custom Search Engine and set as default
+    /// In Private Mode the default engine will be set as Brave Search
+    func migrateDefaultYahooSearchEngines() {
```

Comment:
How hard would be to add a unit test for it? We would need some way to detect if the migrated search engine is a custom one, and handle both jp and non jp

Reply:
You are write lets add test for this.

Reply:
Test is added. Fix at https://github.com/brave/brave-ios/pull/3852/commits/a3be2e9b63840f7f0731dd300e28c919949b528c